src/components/Visualizer.svelte

```diff
@@ -0,0 +1,87 @@
+<script>
+  import { onMount, onDestroy } from 'svelte';
+  import { shaderTexts, gatorToGLSL } from '../modules/utils';
+  import * as glHelpers from '../modules/glHelpers';
+  import Renderer from '../modules/renderer';
+  import teapotObj from '../../assets/models/teapot.obj';
+
+  // If this component is initialized with the gator prob as `false`,
+  // it will just act as vanilla webgl code
+  export let gator = true;
+
+  // These are controlled by svelte
+  let canvas: HTMLCanvasElement;
+  let status = '';
+  let fragText: string;
+  let vertText: string;
+
+  // misc gl stuff
+  const teapotModel = glHelpers.parseOBJ(teapotObj);
+  let renderer: Renderer;
+
+  // Run on component start, not in SSR
+  if (window.routify.inBrowser) {
+    onMount(async () => {
+      // Initialize the GL context. `canvas` is only defined here
+      renderer = new Renderer(canvas);
+      let frag: string;
+      let vert: string;
+      if (gator) {
+        ({ frag, vert } = await gatorToGLSL({frag: fragText, vert: vertText}));
+      } else {
+        frag = fragText;
+        vert = vertText;
+      }
+      renderer.compile(frag, vert, teapotModel);
+      renderer.startRender();
+    });
+  }
+
+  // Run to activate new shader code
+  const compile = async () => {
+    renderer.stopRender();
+    try {
+      let frag: string;
+      let vert: string;
+      if (gator) {
+        ({ frag, vert } = await gatorToGLSL({frag: fragText, vert: vertText}));
```

Comment:
Currently, this recompiles both frag and vert shader code if one of them changes, which requires an additional execution of `gatorc`. This is one of the things I'd like to address before getting this PR in.